core/rawdb/accessors_indexes.go

```diff
@@ -134,14 +135,56 @@ func ReadReceipt(db ethdb.Reader, hash common.Hash, config *params.ChainConfig)
 	if blockHeader == nil {
 		return nil, common.Hash{}, 0, 0
 	}
-	// Read all the receipts from the block and return the one with the matching hash
-	receipts := ReadReceipts(db, blockHash, *blockNumber, blockHeader.Time, config)
-	for receiptIndex, receipt := range receipts {
-		if receipt.TxHash == hash {
-			return receipt, blockHash, *blockNumber, uint64(receiptIndex)
+	blockBody := ReadBody(db, blockHash, *blockNumber)
+	if blockBody == nil {
+		log.Error("Missing body but have receipt", "blockHash", blockHash, "number", blockNumber)
+		return nil, common.Hash{}, 0, 0
+	}
+
+	// Find a match tx and derive receipt fields
+	for txIndex, tx := range blockBody.Transactions {
+		if tx.Hash() == txHash {
```

Comment:
```suggestion
		if tx.Hash() != txHash {
			continue
		}
```
if you invert this, you can un-indent the whole clause below for great good

Reply:
Looks good.
Below lines need to be un-indent, so I'll push a new commit.